ios/StripeTerminalReactNative.swift

```diff
@@ -312,13 +312,16 @@ class StripeTerminalReactNative: RCTEventEmitter, DiscoveryDelegate, BluetoothRe
         let onBehalfOf: String? = params["onBehalfOf"] as? String
         let merchantDisplayName: String? = params["merchantDisplayName"] as? String
         let tosAcceptancePermitted: Bool = params["tosAcceptancePermitted"] as? Bool ?? true
+        let autoReconnectOnUnexpectedDisconnect = params["autoReconnectOnUnexpectedDisconnect"] as? Bool ?? false
 
         let connectionConfig: LocalMobileConnectionConfiguration
         do {
             connectionConfig = try LocalMobileConnectionConfigurationBuilder(locationId: locationId ?? selectedReader.locationId ?? "")
                 .setMerchantDisplayName(merchantDisplayName ?? nil)
                 .setOnBehalfOf(onBehalfOf ?? nil)
                 .setTosAcceptancePermitted(tosAcceptancePermitted)
+                .setAutoReconnectOnUnexpectedDisconnect(autoReconnectOnUnexpectedDisconnect)
+                .setAutoReconnectionDelegate(autoReconnectOnUnexpectedDisconnect ? self : nil)
```

Comment:
not for this PR but we should probably setup the Swift code to use a `connectReader` extension similar to Android's so they're more closely matched 🤔 just noticed this difference (very low pri though 🤷‍♂️)